mozilla-release/mobile/android/tests/cliqz/java/com/cliqz/mobile/test/pages/FreshTabPage.java

```diff
@@ -0,0 +1,196 @@
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.clearData();
+        super.onboardingPref(false);
+        super.defaultBrowserPopupPref(false);
+        super.setUp();
+    }
+
+
+
+    // *** Test Data and/or Requirements ***
+
+    private static final String URL_BAR_TEXT = "Search or enter address"; // Assert Equals
+    private static final String EMPTY_TOP_SITE_TEXT = "Your most visited websites will appear here"; // Assert Equals
+    private static final String GHOSTY_TEXT = "HELLO"; // Assert Equals
+    private static final String MENU_BUTTON_TEXT = "Menu"; // Assert Equals
+
+
+
+    // *** Locators ***
+
+    private static final BySelector TOP_TOOL_BAR = By.res(
+            NativeDriver.APP_PACKAGE + ":id/browser_toolbar");
+    private static final BySelector URL_BAR = By.res(
+            NativeDriver.APP_PACKAGE + ":id/url_bar_title");
+    private static final BySelector URL_BAR_EDIT = By.res(
+            NativeDriver.APP_PACKAGE + ":id/url_edit_text");
+    private static final BySelector URL_BAR_EDIT_CANCEL = By.res(
+            NativeDriver.APP_PACKAGE + ":id/edit_cancel");
+    private static final BySelector GHOSTY_ICON = By.res(
+            NativeDriver.APP_PACKAGE + ":id/ghosty");
+    private static final BySelector TABS_BUTTON = By.res(
+            NativeDriver.APP_PACKAGE + ":id/tabs_counter");
+    private static final BySelector TABS_COUNTER = By.res(
+            NativeDriver.APP_PACKAGE + ":id/counter_text");
+    private static final BySelector MENU_BUTTON = By.res(
+            NativeDriver.APP_PACKAGE + ":id/menu");
+
+
+
+    // *** Page Methods ***
+
+    private UiObject2 getTopToolBar() throws InterruptedException {
+        return driver.getElement(TOP_TOOL_BAR);
+    }
+
+    private UiObject2 getUrlBar() throws InterruptedException {
+        return driver.getElement(getTopToolBar(), URL_BAR);
+    }
+
+    private UiObject2 getUrlBarEdit() throws InterruptedException {
+        return driver.getElement(getTopToolBar(), URL_BAR_EDIT);
+    }
+
+    private UiObject2 getUrlBarEditCancel() throws InterruptedException {
+        return driver.getElement(getTopToolBar(), URL_BAR_EDIT_CANCEL, false);
+    }
+
+    private UiObject2 getGhostyIcon() throws InterruptedException {
+        return driver.getElement(getTopToolBar(), GHOSTY_ICON);
+    }
+
+    private UiObject2 getTabsButton() throws InterruptedException {
+        return driver.getElement(getTopToolBar(), TABS_BUTTON);
+    }
+
+    private int getTabsCount() throws InterruptedException {
+        return Integer.valueOf(
+                driver.getElement(getTabsButton(), TABS_COUNTER).getText()
+        );
+    }
+
+    private UiObject2 getMenuButton() throws InterruptedException {
+        return driver.getElement(getTopToolBar(), MENU_BUTTON);
+    }
+
+
+
+    // *** Test Methods ***
+
+    @Test
+    public void testUrlBar() throws InterruptedException {
```

Comment:
Can we do something to avoid adding `InterruptedException` for every test method?

Reply:
done.